kernel/src/engine/arrow_utils.rs

```diff
@@ -838,42 +838,63 @@ pub(crate) fn to_json_bytes(
 
 #[cfg(test)]
 mod tests {
+    use std::collections::HashMap;
     use std::sync::Arc;
 
+    use crate::parquet::arrow::PARQUET_FIELD_ID_META_KEY;
+
     use crate::arrow::array::{
         Array, ArrayRef as ArrowArrayRef, BooleanArray, GenericListArray, Int32Array, Int32Builder,
         MapArray, MapBuilder, StructArray, StructBuilder,
     };
     use crate::arrow::datatypes::{
-        DataType as ArrowDataType, Field as ArrowField, Fields, Schema as ArrowSchema,
-        SchemaRef as ArrowSchemaRef,
+        DataType as ArrowDataType, Field as ArrowField, Fields as ArrowFields,
+        Schema as ArrowSchema, SchemaRef as ArrowSchemaRef,
     };
     use crate::arrow::{
         array::AsArray,
         buffer::{OffsetBuffer, ScalarBuffer},
     };
 
-    use crate::schema::{ArrayType, DataType, MapType, StructField, StructType};
+    use crate::schema::{
+        ArrayType, ColumnMetadataKey, DataType, MapType, MetadataValue, StructField, StructType,
+    };
+    use crate::table_features::ColumnMappingMode;
     use crate::utils::test_utils::assert_result_error_with_message;
 
     use super::*;
 
-    fn nested_parquet_schema() -> ArrowSchemaRef {
-        Arc::new(ArrowSchema::new(vec![
-            ArrowField::new("i", ArrowDataType::Int32, false),
-            ArrowField::new(
-                "nested",
-                ArrowDataType::Struct(
-                    vec![
-                        ArrowField::new("int32", ArrowDataType::Int32, false),
-                        ArrowField::new("string", ArrowDataType::Utf8, false),
-                    ]
-                    .into(),
-                ),
-                false,
-            ),
-            ArrowField::new("j", ArrowDataType::Int32, false),
-        ]))
+    fn column_mapping_cases() -> [ColumnMappingMode; 1] {
+        [ColumnMappingMode::Name]
+    }
+
+    type NameFn = Box<dyn Fn(usize) -> String>;
+    type MetadataFn = Box<dyn Fn(usize) -> HashMap<String, MetadataValue>>;
+    /// Returns closures for generating column names and metadata based on the mapping mode.
+    ///
+    /// # Returns
+    /// A tuple of (logical_name_fn, physical_name_fn, parquet_name_fn, metadata_fn):
+    /// - logical_name_fn: Generates the logical name for a field given its id.
+    ///   This is "logical-{fieldId}".
+    /// - physical_name_fn: Generates the physical name for field given its id.
+    ///   This is "physical-{fieldId}".
+    /// - parquet_name_fn: Generates the name that should be written to parquet from the field id.
+    ///   This is the physical name for Id/Name modes, and logical name for None mode
+    /// - metadata_fn: Generates the column mapping metadata for a logical struct field given the field id.
+    fn names_and_metadata(mode: ColumnMappingMode) -> (NameFn, NameFn, NameFn, MetadataFn) {
+        let logical_name = |i: usize| format!("logical-{i}");
+        let physical_name = |i: usize| format!("physical-{i}");
```

Comment:
These don't vary based on the mode. Do we expect something in the future to have them need to be dependent? If not I think I'd prefer to move these out to top-level functions that we can just call directly. 

In fact, I think the `metadata_fn` also doesn't vary? In which case can we just have 4 top level functions, 3 of which just take `i: usize`, and then one that's  `fn parquet_name(i: usize, mode: ColumnMappingMode)`?

Reply:
Yeah I was thinking this could've been randomized, but whether the column is called `logical_1` or `0x3467065487`, it doesn't change the test much.


done 👍